lib/reek/smells/smell_detector.rb

```diff
@@ -104,16 +93,23 @@ def examine(context)
         self.smells_found += sm
       end
 
-      def enabled_for?(context)
-        enabled? && config_for(context)[SmellConfiguration::ENABLED_KEY] != false
+      def report_on(report)
+        smells_found.each { |smell| smell.report_on(report) }
       end
 
       def exception?(context)
         context.matches?(value(EXCLUDE_KEY, context, DEFAULT_EXCLUDE_SET))
       end
 
-      def report_on(report)
-        smells_found.each { |smell| smell.report_on(report) }
+      protected
+
+      # NOTE: Writers cannot be private in Ruby < 2.2
+      attr_writer :smells_found
```

Comment:
>> Because that causes self.smells_found = 0 not to work in Ruby < 2.2.

I don't follow :(

>>   ( $ (master)) grep private_attr_writer lib/ -r
lib//reek/configuration/app_configuration.rb:      private_attr_writer :directory_directives, :default_directive, :excluded_paths
lib//reek/context/code_context.rb:      private_attr_writer :tracked_visibility

Those writer work with all ruby versions we run on travis.